cdap-cli/src/main/java/co/cask/cdap/cli/command/GetStreamStatsCommand.java

```diff
@@ -0,0 +1,372 @@
+      processorMap.put(hiveColumnName, getProcessorsForType(fieldSchema.getType(), fieldSchema.getUnionSchemas()));
+    }
+
+    // get a list of stream events and calculates various statistics about the events
+    String timestampCol = getTimestampHiveColumn(streamId);
+    ListenableFuture<ExploreExecutionResult> resultsFuture = queryClient.execute(
+      "SELECT * FROM cdap_stream_" + streamId
+        + " WHERE " + timestampCol + " BETWEEN " + startTime + " AND " + endTime
+        + " LIMIT " + limit);
+    ExploreExecutionResult results = resultsFuture.get(1, TimeUnit.MINUTES);
+    List<ColumnDesc> schema = results.getResultSchema();
+
+    // apply StatsProcessors to every element in every row
+    while (results.hasNext()) {
+      QueryResult row = results.next();
+      for (int i = 0; i < row.getColumns().size(); i++) {
+        Object column = row.getColumns().get(i);
+        ColumnDesc columnDesc = schema.get(i);
+        String columnName = columnDesc.getName();
+        if (isUserHiveColumn(streamId, columnName)) {
+          Set<StatsProcessor> processors = processorMap.get(columnName);
+          if (processors != null) {
+            for (StatsProcessor processor : processors) {
+              processor.process(column);
+            }
+          }
+        }
+      }
+    }
+
+    // print report
+    for (ColumnDesc columnDesc : schema) {
+      if (isUserHiveColumn(streamId, columnDesc.getName())) {
+        String truncatedColumnName = getTruncatedColumnName(streamId, columnDesc.getName());
+        output.printf("column: %s, type: %s", truncatedColumnName, columnDesc.getType());
+        output.println();
+        Set<StatsProcessor> processors = processorMap.get(columnDesc.getName());
+        if (processors != null && !processors.isEmpty()) {
+          for (StatsProcessor processor : processors) {
+            processor.printReport(output);
+          }
+          output.println();
+        } else {
+          output.println("No statistics available");
+          output.println();
+        }
+      }
+    }
+  }
+
+  private String getTruncatedColumnName(String streamId, String hiveColumnName) {
+    String hiveTableName = getHiveTableName(streamId);
+    String hiveTablePrefix = hiveTableName + ".";
+    if (hiveColumnName.startsWith(hiveTablePrefix)) {
+      return hiveColumnName.substring(hiveTablePrefix.length());
+    }
+    return hiveColumnName;
+  }
+
+  private String getTimestampHiveColumn(String streamId) {
+    return cdapSchemaColumName2HiveColumnName(streamId, "ts");
+  }
+
+  private boolean isUserHiveColumn(String streamId, String hiveColumName) {
+    // TODO: hardcoded
+    return !cdapSchemaColumName2HiveColumnName(streamId, "ts").equals(hiveColumName)
+      && !cdapSchemaColumName2HiveColumnName(streamId, "headers").equals(hiveColumName);
+  }
+
+  private String getHiveTableName(String streamId) {
+    return "cdap_stream_" + streamId;
```

Comment:
Is "cdap" prefix configurable? If this comes from root.namespace property then this would break. Let's make sure we do this consistently for CLI and streams code.